staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -362,8 +362,10 @@ func (e *Store) WaitForInitialized(ctx genericapirequest.Context, obj runtime.Ob
 		select {
 		case event, ok := <-ch:
 			if !ok {
-				// TODO: should we just expose the partially initialized object?
-				return nil, kubeerr.NewServerTimeout(e.QualifiedResource, "create", 0)
+				return latest,
```

Comment:
NewTimeoutError already indicates that the client may retry, but the operation should still be continuing.  So this code should be returning `NewTimeoutError`, not `NewServerTimeout`, and you should check for `NewTimeoutError`.